sources/web/datalab/sockets.ts

```diff
@@ -165,10 +166,11 @@ function socketHandler(socket: SocketIO.Socket) {
 
 export function init(settings: common.Settings): void {
   var io = socketio(String(settings.socketioPort), {
+    path: path_.join(settings.datalabBasePath, '/socket.io'),
```

Comment:
You already ensured that datalabBasePath ends with a slash (in settings.ts), so you don't need one here. Likewise in other places where you append to datalabBasePath.